update_agol_vehicles_pallet.py

```diff
@@ -0,0 +1,201 @@
+'''
+update_agol_vehicles_pallet.py:
+Automates pulling csvs from an FTP folder and updating a hosted feature
+service with the contents of the latest one.
+'''
+
+import datetime
+import os
+import sys
+
+from pathlib import Path
+from urllib.error import HTTPError
+
+import arcgis
+import arcpy
+import pysftp
+
+from forklift.models import Pallet
+
+import fleetshare_secrets as secrets
+
+
+def get_latest_csv(temp_csv_dir, log, previous_days=-1):
+    '''
+    Returns the Path object and date of the latest 'vehicle_data_*.csv' file in
+    temp_csv_dir. Will fail if limit_three_days is True and the date on the
+    latest csv does not fall within the three preceding days.
+    '''
+
+    #: get list of csvs
+    temp_dir_path = Path(temp_csv_dir)
+    csvs = sorted(temp_dir_path.glob('vehicle_data_*.csv'))
+
+    #: The last of the sorted list of csvs should be the latest
+    try:
+        latest_csv = csvs[-1]
+    except IndexError as e:
+        log.error('Can\'t get last "vehicle_data_*.csv" file- are there any'
+                  ' csv files?')
+        raise e
+
+    #: Pull the date out of vehicle_data_yyyymmdd.csv to check recency
+    date_string = str(latest_csv).rsplit('_')[-1].split('.')[0]
+    try:
+        csv_datetime = datetime.date(int(date_string[:4]),
+                                     int(date_string[4:6]),
+                                     int(date_string[6:]))
+    except ValueError as e:
+        log.error(f'Can\'t parse date from last csv: {latest_csv}')
+        raise e
+
+    #: Only continue if the latest is within specified number of days
+    today = datetime.date.today()
+    previous_dates = [today - datetime.timedelta(days=i)
+                      for i in range(previous_days + 1)]
+    if previous_days > 0 and csv_datetime not in previous_dates:
+        message = (f'Latest csv "{latest_csv}" not within {previous_days} days'
+                   f' of today ({today})')
+        log.info(message)
+        sys.exit(message)
+
+    return latest_csv, date_string
+
+
+def get_map_layer(project_path, fc_to_add, log):
```

Comment:
**question**: is there a reason for not making these methods on the class? You wouldn't need to pass log around if they were.

Reply:
You are free to add methods willy nilly. I don't see where you using a constructor method. Why would it need to be overridden?

Reply:
Maybe it's my old C#/java from college coming out, but usually when I create a class I like to instantiate the variables/properties in the constructor. It helps me keep track of what I can expect to use elsewhere. This stemmed from thinking that if I was going to make them class methods I could access other data directly rather than passing through the method call. Maybe that's thinking too deep about it?

Reply:
It might be best to talk this out over a liveshare + audio or meet to get the full picture.

Reply:
Ok, they don't try to use any other instance variables beyond `self.log`, but they are now class methods. How does this look to you?